Sources/Filters/General/ClipClosedSurface/ccsEdgeLocator.js

```diff
@@ -0,0 +1,55 @@
+export default class CCSEdgeLocator {
+  constructor() {
+    this._edgeMap = new Map();
+  }
+
+  initialize() {
+    this._edgeMap.clear();
+  }
+
+  insertUniqueEdge(i0, i1, i) {
+    // Ensure consistent ordering of edge
+    if (i1 < i0) {
+      // eslint-disable-next-line no-param-reassign
+      [i0, i1] = [i1, i0];
+    }
+
+    // Generate a integer key, try to make it unique
+    // eslint-disable-next-line no-bitwise
+    const key = (i1 << 16) ^ i0;
```

Comment:
key does not have to be that complicated, it could have been simply `[i0, i1]`. You would then not need `next` because you know it is unique.